File: Frontend/src/api/auditoria.ts
import { apiFetch } from "./client";

export type EventoAuditoria = {
  id_auditoria: number;
  id_usuario: number | null;
  username: string;
  accion: "ANULAR_PEDIDO" | "BORRAR_DEFINITIVO" | "AUTORIZAR_DESCUENTO" | "LIMPIAR_TRANSACCIONES";
  entidad: string;
  id_entidad: number | null;
  detalle: string | null;
  fecha: string;
};

export const listarAuditoria = (token: string | null, limite = 200) =>
  apiFetch<EventoAuditoria[]>(`/auditoria/?limite=${limite}`, { token });

File: Backend/app/routers/auditoria.py
from fastapi import APIRouter, Depends
from sqlalchemy import text

from app.database import engine
from app.rbac import Rol, require_role

router = APIRouter(prefix="/auditoria", tags=["Auditoría"])


@router.get("/")
def listar(limite: int = 200, _: dict = Depends(require_role(Rol.ADMIN))):
    with engine.connect() as conn:
        filas = conn.execute(
            text("""
                SELECT id_auditoria, id_usuario, username, accion, entidad, id_entidad, detalle, fecha
                FROM auditoria
                ORDER BY fecha DESC
                LIMIT :lim
            """),
            {"lim": min(max(limite, 1), 1000)},
        )
        return [dict(f._mapping) for f in filas]
